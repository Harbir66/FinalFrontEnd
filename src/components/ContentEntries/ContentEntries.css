.main-entries {
  width: 60vw;
  height: 10vh;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
}

.column-names-entries {
  width: 60vw;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0;
  background-color: white;
}

.item-entries {
  width: 6vw;
  margin: 0;
}

.buttons-entries {
  width: 10vw;
  margin: 0 0 0 1vw;
}

.buttons-entries > button {
  background: none;
  border: none;
  outline: none;
  cursor: pointer;
  height: 2vh;
}
