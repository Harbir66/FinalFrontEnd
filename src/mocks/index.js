export const mockCollection = [
  {
    id: 1,
    name: 'collection1',
    fields: 3,
    values: 3,
    allFields: ['field1', 'field2', 'field3'],
  },
  {
    id: 2,
    name: 'collection2',
    fields: 3,
    values: 3,
    allFields: ['field1', 'field2', 'field3'],
  },
];

export const mockHeading = 'Heading';

export const mockFields = ['field1', 'field2', 'field3'];
