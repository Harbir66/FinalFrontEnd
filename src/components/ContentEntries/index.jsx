import React from 'react';
import './ContentEntries.css';
import PropTypes from 'prop-types';
import { stringToArray } from '../../utils/common';
import editIcon from '../../assets/user-edit-text-message-note@3x.png';
import deleteIcon from '../../assets/trash-delete-recycle-bin-bucket-waste@3x.png';

function ContentEntries({ entry, fields, index }) {
  const firstFourFields = fields.slice(0, 4);
  firstFourFields.unshift('id');
  const values = stringToArray(entry);

  const newValues = values.reduce(
    (acc, value) => {
      const temp = value.split(':');
      return { ...acc, [temp[0]]: temp[1] };
    },
    { id: index }
  );

  return (
    <div className="main-entries">
      <div className="column-names-entries">
        {firstFourFields.map((field) => (
          <div className="item-entries" key={field}>
            {newValues[field]}
          </div>
        ))}
      </div>
      <div className="buttons-entries">
        <button type="button">
          <img src={editIcon} alt="edit-icon" />
        </button>
        <button type="button">
          <img src={deleteIcon} alt="delete-icon" />
        </button>
      </div>
    </div>
  );
}

export default ContentEntries;

ContentEntries.propTypes = {
  entry: PropTypes.string.isRequired,
  fields: PropTypes.arrayOf(PropTypes.string).isRequired,
  index: PropTypes.number.isRequired,
};
