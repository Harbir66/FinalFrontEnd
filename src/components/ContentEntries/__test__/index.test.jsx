import React from 'react';
import { render } from '@testing-library/react';
import ContentEntries from '..';
import { mockFields } from '../../../mocks';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

describe('ContentEntries', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <ContentEntries
        entry="mockEntry"
        fields={mockFields}
        index={1}
        handleDeleteEntries={jest.fn()}
        selectedCollectionId={2}
        setEntries={jest.fn()}
        selectedCollectionAllFields={mockFields}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
