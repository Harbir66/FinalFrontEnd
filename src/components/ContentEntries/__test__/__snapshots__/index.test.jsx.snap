// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentEntries should render correctly 1`] = `
<DocumentFragment>
  <div
    class="main-entries"
  >
    <div
      class="column-names-entries"
    >
      <div
        class="item-entries"
      >
        1
      </div>
      <div
        class="item-entries"
      />
      <div
        class="item-entries"
      />
      <div
        class="item-entries"
      />
    </div>
    <div
      class="buttons-entries"
    >
      <button
        type="button"
      >
        <img
          alt="edit-icon"
          src="user-edit-text-message-note@3x.png"
        />
      </button>
      <button
        type="button"
      >
        <img
          alt="delete-icon"
          src="trash-delete-recycle-bin-bucket-waste@3x.png"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
